x-pack/plugins/apm/server/lib/status_check/agent_check.ts

```diff
@@ -19,11 +21,17 @@ export async function getAgentStatus({ setup }) {
       size: 0,
       query: {
         bool: {
-          should: [
-            { term: { [PROCESSOR_NAME]: 'error' } },
-            { term: { [PROCESSOR_NAME]: 'transaction' } },
-            { term: { [PROCESSOR_NAME]: 'metric' } },
-            { term: { [PROCESSOR_NAME]: 'sourcemap' } }
+          filter: [
+            {
+              terms: {
+                [PROCESSOR_EVENT]: [
+                  'error',
+                  'transaction',
+                  'metric',
+                  'sourcemap'
+                ]
+              }
```

Comment:
Again: converting a `should` clause to the faster `filter` clause. `filter` is cachable and doesn't calculate a score.